internal/bminventory/inventory.go

```diff
@@ -1981,46 +1981,58 @@ func (b *bareMetalInventory) noneHaModeClusterUpdateValidations(cluster *common.
 	return nil
 }
 
-func (b *bareMetalInventory) validateAndUpdateClusterParams(ctx context.Context, params *installer.UpdateClusterParams) (installer.UpdateClusterParams, error) {
+func (b *bareMetalInventory) v2NoneHaModeClusterUpdateValidations(cluster *common.Cluster, params installer.V2UpdateClusterParams) error {
```

Comment:
why do we need to rename this function? Why not leave it as is and just change the input params to be V2?

Reply:
This is the single function for which I have made a copy.
noneHaModeClusterUpdateValidations  updates `HostsRoles` which you do not have in `V2UpdateClusterParams`:
https://github.com/openshift/assisted-service/blob/bb4e8cf7fa547570ca84792d4eae72fa6a9f3f46/internal/bminventory/inventory.go#L1968-L1982

Therefore I needed a copy function that does everything else the same way (e.g. return codes for the other checks) but doesn't have that check.

Reply:
and btw the `validateAndUpdateClusterParams` was not renamed. the GitHub diff is misleading here.
see the `validateAndUpdateClusterParams` function below at line 1996. the change there was `UpdateClusterParams` --> `V2UpdateClusterParams`